apps/evm_family/evm_pub_key.c

```diff
@@ -343,30 +400,44 @@ void evm_get_pub_keys(evm_query_t *query) {
   // button pressed on PIN/Passphrase entry
   if (!reconstruct_seed_flow(query->get_public_keys.initiate.wallet_id,
                              &seed[0])) {
-    memzero(seed, sizeof(seed));
     // TODO: Handle error case reporting to host
+    memzero(seed, sizeof(seed));
     return;
   }
 
   delay_scr_init(ui_text_processing, DELAY_SHORT);
 
-  bool status = get_public_keys(init_req->derivation_paths,
-                                init_req->derivation_paths_count,
-                                seed,
-                                public_keys);
+  bool status = fill_public_keys(init_req->derivation_paths,
+                                 seed,
+                                 public_keys,
+                                 init_req->derivation_paths_count);
 
+  // Clear seed as soon as it is not needed
   memzero(seed, sizeof(seed));
-  if (status) {
-    status =
-        send_public_keys(query, public_keys, init_req->derivation_paths_count);
-  } else {
+
+  if (!status) {
     // send unknown error; do not know failure reason
     evm_send_error(ERROR_COMMON_ERROR_UNKNOWN_ERROR_TAG, 1);
+    return;
   }
 
-  if (status) {
-    delay_scr_init(ui_text_check_cysync_app, DELAY_TIME);
+  if (EVM_QUERY_GET_USER_VERIFIED_PUBLIC_KEY_TAG == which_request) {
+    char address[100] = "";
+    byte_array_to_hex_string(
+        public_keys[0], sizeof(public_keys[0]), address, sizeof(address));
```

Comment:
need to encode the ethereum address. ref: `hdnode_get_ethereum_pubkeyhash` and `ethereum_address_checksum`